src/core.jl

```diff
@@ -50,6 +51,56 @@ function transform!(context::MD5_CTX)
     @inbounds context.state .+= [a,b,c,d]
 end
 
+@generated function transform_unrolled!(context::MD5_CTX)
+    ret = quote
+        pbuf = buffer_pointer(context)
+    end
+    ex  = quote
+        A = context.state[1]
```

Comment:
add `@inbounds` to all of these?

Reply:
The whole body is `@inbounds`.

Reply:
Or does it propagate correctly from the last line?

Reply:
It just does not propagate into function calls. But I think you are still right it is better style to be explict, if something is refactored into a function.